README.md

```diff
@@ -379,6 +379,34 @@ Succeeded   : 50
 
 </details>
 
+## Design
+
+### High-level Decisions
+
+- **Nextflow Workflow**
+  - Many QC tests will require the output from one or more commands using external (_i.e._ non-Python) programs. While we can run external commands in Python (_e.g. using the `subprocess` module), we want to avoid re-inventing a workflow management system (WMS). Instead, we should leverage an existing WMS. For the purposes of this POC, we opted for Nextflow since we have the compute infrastructure set up. That said, a future iteration could use the Common Workflow Language (whose reference implementation is in Python) and/or the Task Execution Service (TES).
```

Comment:
That's pretty interesting to leverage CWL with the combination of leveraging TES.